### PR TITLE
[Photos] Add support for Xcode 15 beta 8

### PR DESCRIPTION
**Pull request by haritha-mohan**

### CODE REVIEW

File: src/photos.cs [hunk 1]
```diff
@@ -795,6 +796,19 @@ interface PHContentEditingOutput : NSCoding, NSSecureCoding {
...
+		UTType DefaultRenderedContentType { get; }
+
+		[TV(17, 0), Mac(14, 0), iOS(17, 0), MacCatalyst(17, 0)]
+		[Export("supportedRenderedContentTypes", ArgumentSemantic.Copy)]
+		UTType[] SupportedRenderedContentTypes { get; }
+
+		[TV(17, 0), Mac(14, 0), iOS(17, 0), MacCatalyst(17, 0)]
+		[Export("renderedContentURLForType:error:")]
+		[return: NullAllowed]
+		NSUrl RenderedContentURLForType(UTType type, [NullAllowed] out NSError error);
```
Comment (rolfbjarne) [inline src/photos.cs:811]:
```suggestion
		NSUrl GetRenderedContentUrl (UTType type, [NullAllowed] out NSError error);
```

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: src/photos.cs (hunk 1 above)
Comment (haritha-mohan) [reply src/photos.cs]: you were faster than auto format haha

File: src/photos.cs [hunk 2]
```diff
@@ -795,6 +796,19 @@ interface PHContentEditingOutput : NSCoding, NSSecureCoding {
...
+		UTType DefaultRenderedContentType { get; }
+
+		[TV (17, 0), Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+		[Export ("supportedRenderedContentTypes", ArgumentSemantic.Copy)]
+		UTType [] SupportedRenderedContentTypes { get; }
+
+		[TV (17, 0), Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+		[Export ("renderedContentURLForType:error:")]
+		[return: NullAllowed]
+		NSUrl RenderedContentURLForType (UTType type, [NullAllowed] out NSError error);
```
Comment (haritha-mohan) [inline src/photos.cs:811]:
```suggestion
		NSUrl GetRenderedContentUrl (UTType type, [NullAllowed] out NSError error);
```

File: src/Photos/Enums.cs [hunk 3]
```diff
@@ -259,6 +259,9 @@ public enum PHAssetResourceType : long {
 		[iOS (13, 0), TV (13, 0)]
 		[MacCatalyst (13, 1)]
 		AdjustmentBaseVideo = 12,
+		[iOS (17, 0), TV (10, 0)]
```
Comment (tj-devel709) [inline src/Photos/Enums.cs:262]: Should this be TV 10? Maybe version 17 would be safer @mandel-macaque

File: src/Photos/Enums.cs [hunk 4]
```diff
@@ -259,6 +259,9 @@ public enum PHAssetResourceType : long {
...
+		[MacCatalyst (17, 1), Mac (14, 0)]
```
Comment (tj-devel709) [inline src/Photos/Enums.cs:263]: Is Catalyst 17.1 or 17?

**Review by tj-devel709 — CHANGES_REQUESTED** [review]
Just want to check if Apple is being fishy with the versioning of PhotoProxy

File: src/Photos/Enums.cs (hunk 3 above)
Comment (mandel-macaque) [reply src/Photos/Enums.cs:262]: I agree on that, also is definitely a MacCatalyst 17, not 17.1

**Review by mandel-macaque — CHANGES_REQUESTED** [review]
Needs fixing as per tj reviews.

**Review by dalexsoto — APPROVED** [review]

File: src/Photos/Enums.cs (hunk 3 above)
Comment (haritha-mohan) [inline src/Photos/Enums.cs:262]:
```suggestion
		[iOS (17, 0), TV (17, 0)]
```

File: src/Photos/Enums.cs (hunk 4 above)
Comment (haritha-mohan) [inline src/Photos/Enums.cs:263]:
```suggestion
		[MacCatalyst (17, 0), Mac (14, 0)]
```

File: src/Photos/Enums.cs (hunk 3 above)
Comment (haritha-mohan) [reply src/Photos/Enums.cs]: @tj-devel709 good catch, addressed!

**Review by tj-devel709 — APPROVED** [review]

**Review by mandel-macaque — APPROVED** [review]

**Pull request merged**